casanovo/data/ms_io.py

```diff
@@ -143,10 +230,95 @@ def set_ms_run(self, peak_filenames: List[str]) -> None:
             )
             self._run_map[filename] = i
 
+    def get_report_dict(self) -> Optional[Dict]:
+        """
+        Generate sequencing run report
+
+        Parameters
+        ----------
+            score_bins: List[float], Optional
+                Confidence scores for creating confidence CMF, see getScoreBins
+
+        Returns:
+            report_gen: Dict
+                Generated report represented as a dictionary, or None if no
+                sequencing predictions were logged
+        """
+        results_table = DataFrame(
+            {
+                "sequence": [psm[0] for psm in self.psms],
+                "score": [psm[2] for psm in self.psms],
+            }
+        )
+
+        if results_table.empty:
+            return None
+
+        peptide_lengths = get_peptide_lengths(results_table)
+        return {
+            "num_spectra": len(results_table),
+            "score_bins": get_score_bins(results_table, self.score_bins),
+            "max_sequence_length": int(np.max(peptide_lengths)),
```

Comment:
**nitpick:** Should be possible to get this for free from the peptide lengths histogram (if the bins are sorted).